include/aspect/particle/property/cpo_elastic_tensor.h

```diff
@@ -0,0 +1,199 @@
+           *
+           * @param [in] position The current particle position.
+           *
+           * @param [in] solution The values of the solution variables at the
+           * current particle position.
+           *
+           * @param [in] gradients The gradients of the solution variables at
+           * the current particle position.
+           *
+           * @param [in,out] particle_properties The properties of the particle
+           * that is updated within the call of this function.
+           */
+          void
+          update_one_particle_property (const unsigned int data_position,
+                                        const Point<dim> &position,
+                                        const Vector<double> &solution,
+                                        const std::vector<Tensor<1,dim>> &gradients,
+                                        const ArrayView<double> &particle_properties) const override;
+
+          /**
+           * This implementation tells the particle manager that
+           * we need to update particle properties every time step.
+           */
+          UpdateTimeFlags
+          need_update () const override;
+
+          /**
+           * Return which data has to be provided to update the property.
+           * The integrated strains needs the gradients of the velocity.
+           */
+          UpdateFlags
+          get_needed_update_flags () const override;
+
+          /**
+           * Set up the information about the names and number of components
+           * this property requires.
+           *
+           * @return A vector that contains pairs of the property names and the
+           * number of components this property plugin defines.
+           */
+          std::vector<std::pair<std::string, unsigned int>>
+          get_property_information() const override;
+
+          /**
+           * Loads particle data into variables
+           */
+          static
+          SymmetricTensor<2,6>
+          get_elastic_tensor(unsigned int cpo_index,
+                             const ArrayView<double> &data);
+
+          /**
+           * Stores information in variables into the data array
+           */
+          static
+          void
+          set_elastic_tensor(unsigned int cpo_data_position,
+                             const ArrayView<double> &data,
+                             SymmetricTensor<2,6> &elastic_tensor);
+
+          /**
+           * Declare the parameters this class takes through input files.
+           */
+          static
+          void
+          declare_parameters (ParameterHandler &prm);
+
+          /**
+           * Read the parameters this class declares from the parameter file.
+           */
+          void
+          parse_parameters (ParameterHandler &prm) override;
+
+        private:
+          unsigned int cpo_data_position;
+
+          SymmetricTensor<2,6> stiffness_matrix_olivine;
```

Comment:
Is this an appropriate time to generalise to std::vector<SymmetricTensor<2,6>> stiffness_matrices?

I realise this might be a lot of work, so an alternative might be to rename this .h and .cc files and the class to better describe what they do. Maybe `OlivineEnstatiteCPO`?

Reply:
as discussed, let's leave the name for now and generatlize it in a follow up pull request.